src/Symfony/Component/DomCrawler/Tests/ImageTest.php

```diff
@@ -25,6 +25,17 @@ public function testConstructorWithANonImgTag()
         new Image($dom->getElementsByTagName('div')->item(0), 'http://www.example.com/');
     }
 
+    /**
+     * @requires PHP 8.4
+     */
+    public function testConstructorWithANonImgTagFromHTMLDocument()
+    {
+        $this->expectException(\LogicException::class);
```

Comment:
this should be move after the creation of the DOM Document, to ensure that the expected exception cannot happen during that parsing.

Reply:
same in other exception tests